blocks/header/header.js

```diff
@@ -116,34 +140,79 @@ export default async function decorate(block) {
-      navSection.addEventListener('click', () => {
-        if (isDesktop.matches) {
-          const expanded = navSection.getAttribute('aria-expanded') === 'true';
-          toggleAllNavSections(navSections);
-          navSection.setAttribute('aria-expanded', expanded ? 'false' : 'true');
-        }
-      });
+      if (navSection.querySelector('ul')) {
+        const textWrapper = document.createElement('a');
+        textWrapper.classList.add('nav-drop-text');
+        textWrapper.append(navSection.firstChild);
+        navSection.prepend(textWrapper);
+        navSection.classList.add('nav-drop');
+      }
+
+      navSection.addEventListener('click', () => toggleSubNav(navSection, navSections));
+    });
+    const defaultContentWrapper = navSections.querySelector('.default-content-wrapper');
+    const logoButton = defaultContentWrapper.querySelector('.default-content-wrapper .button');
+    if (logoButton) {
+      logoButton.className = 'nav-logo';
+      defaultContentWrapper.prepend(logoButton);
+      const buttonContainer = defaultContentWrapper.querySelector('.button-container');
+      mobileLogoWrapper.appendChild(logoButton.cloneNode(true));
+      buttonContainer.remove();
+    }
+  }
+
+  const navTools = nav.querySelector('.nav-tools');
+  if (navTools) {
+    const toolsWrapper = navTools.querySelector('ul');
+    toolsWrapper.classList.add('default-content-wrapper');
+    navTools.append(toolsWrapper);
+    navTools.firstElementChild.remove();
+    toolsWrapper.querySelectorAll('li').forEach((list) => {
+      list.addEventListener('click', redirectPage);
     });
   }
 
+  if (navSections && navTools) {
+    const navLinksWrapper = document.createElement('div');
+    navLinksWrapper.classList.add('nav-link-section');
+    navLinksWrapper.append(navSections, navTools);
+    nav.append(navLinksWrapper);
+  }
+
   // hamburger for mobile
   const hamburger = document.createElement('div');
   hamburger.classList.add('nav-hamburger');
   hamburger.innerHTML = `<button type="button" aria-controls="nav" aria-label="Open navigation">
-      <span class="nav-hamburger-icon"></span>
+      <span class="nav-hamburger-icon"></a>
     </button>`;
   hamburger.addEventListener('click', () => toggleMenu(nav, navSections));
   nav.prepend(hamburger);
   nav.setAttribute('aria-expanded', 'false');
   // prevent mobile nav behavior on window resize
   toggleMenu(nav, navSections, isDesktop.matches);
-  isDesktop.addEventListener('change', () => toggleMenu(nav, navSections, isDesktop.matches));
+  isDesktop.addEventListener('change', () => {
+    toggleMenu(nav, navSections, isDesktop.matches);
+  });
 
   const navWrapper = document.createElement('div');
   navWrapper.className = 'nav-wrapper';
-  navWrapper.append(nav);
+  navWrapper.append(mobileLogoWrapper, nav);
   block.append(navWrapper);
+
+  checkForActiveLink(navSections);
+
+  const observer = new IntersectionObserver((entries) => {
```

Comment:
It's not showing the top image (2 red and 2 black strokes) when the initial resolution is desktop.

Reply:
This expected behaviour